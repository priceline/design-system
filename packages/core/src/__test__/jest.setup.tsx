import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
