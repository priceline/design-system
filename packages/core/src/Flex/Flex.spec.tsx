import { render, screen } from '../__test__/testing-library'
import { Flex } from './Flex'

describe('Flex', () => {
  test('renders', () => {
    const { asFragment } = render(<Flex />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('alignItems prop', () => {
    render(<Flex alignItems='center' data-testid='flex' />)
    const element = screen.getByTestId('flex')
    expect(element).toHaveStyleRule('align-items', 'center')
  })

  test('justifyContent prop', () => {
    render(<Flex justifyContent='space-between' data-testid='flex' />)
    const element = screen.getByTestId('flex')
    expect(element).toHaveStyleRule('justify-content', 'space-between')
  })

  test('flexWrap prop', () => {
    render(<Flex flexWrap='wrap' data-testid='flex' />)
    const element = screen.getByTestId('flex')
    expect(element).toHaveStyleRule('flex-wrap', 'wrap')
  })
})
