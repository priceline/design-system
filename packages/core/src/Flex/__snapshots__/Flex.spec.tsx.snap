// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex renders 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    class="c0"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
